crates/cargo-test-support/src/registry.rs

```diff
@@ -611,16 +638,72 @@ impl HttpServer {
                     self.dl(&req)
                 }
             }
+            // publish
+            ("put", ["api", "v1", "crates", "new"]) => {
+                if !authorized(true) {
+                    return self.unauthorized(req)
+                }
+                if let Some(body) = &req.body {
+                    // Get the metadata of the package
+                    let (len, remaining)  = body.split_at(4);
+                    let json_len = u32::from_le_bytes(len.try_into().unwrap());
+                    let (json, remaining) = remaining.split_at(json_len as usize);
+                    let new_crate = serde_json::from_slice::<crates_io::NewCrate>(json).unwrap();
+                    // Get the `.crate` file
+                    let (len, remaining) = remaining.split_at(4);
+                    let file_len = u32::from_le_bytes(len.try_into().unwrap());
+                    let (file, _remaining) = remaining.split_at(file_len as usize);
+
+                    // Write the `.crate`
+                    let dst = self.dl_path.join(&new_crate.name).join(&new_crate.vers).join("download");
+                    t!(fs::create_dir_all(dst.parent().unwrap()));
+                    let mut f = t!(File::create(&dst));
+                    t!(f.write_all(file));
+
+                    let deps = new_crate.deps.iter().map(|dep| {
+                        serde_json::json!({
+                            "name": dep.name,
+                            "req": dep.version_req,
+                            "features": dep.features,
+                            "default_features": true,
+                            "target": dep.target,
+                            "optional": dep.optional,
+                            "kind": dep.kind,
+                            "registry": dep.registry,
+                            "package": dep.explicit_name_in_toml,
+                        })
+                    }).collect::<Vec<_>>();
+
+                    let line = create_index_line(
+                        serde_json::json!(new_crate.name),
+                        &new_crate.vers,
+                        deps,
+                        &cksum(file),
+                        new_crate.features,
+                        false,
+                        new_crate.links,
+                        None,
+                    );
+
+                    write_to_index(&self.registry_path, &new_crate.name, line, false);
+
+                    self.ok(&req)
+                } else {
+                    Response {
+                        code: 400,
+                        headers: vec![],
+                        body: b"The request was missing a body".to_vec(),
+                    }
+                }
+            }
             // The remainder of the operators in the test framework do nothing other than responding 'ok'.
             //
             // Note: We don't need to support anything real here because the testing framework publishes crates
```

Comment:
This note is out of date

Reply:
Since this comment is out of date do you think it would be best to implement all of the endpoints in one PR?

Reply:
I do not think you need to impl the other endpoints, but the text of `note` needs to be updated.